### ISSUE TITLE
First build not Universal, second build is

### ISSUE THREAD

**Issue by PenelopeFudd** [universal-ctags/ctags#3095]
Hi;

TL;DR: A manual install of ctags may be hidden by bash's command cache; type `hash -r` to clear it.  Also, don't have two versions of ctags in your path.

### What happened?

On a fresh Ubuntu 18.04 system, I install Universal-ctags with the instructions from the README:
```
    $ git clone https://github.com/universal-ctags/ctags.git
    $ cd ctags
    $ ./autogen.sh
    $ ./configure --prefix=/where/you/want # defaults to /usr/local
    $ make
    $ sudo make install
```

When I ran `ctags --version`, it printed this:

```
Exuberant Ctags 5.9~svn20110310, Copyright (C) 1996-2009 Darren Hiebert
  Addresses: <dhiebert@users.sourceforge.net>, http://ctags.sourceforge.net
  Optional compiled features: +wildcards, +regex
```
I was greatly puzzled, and tried to figure out how I could be running the wrong ctags after just building and installing it!
So, I did `git --reset hard` the repository and reran the steps, and *this* time it worked correctly: 
```
Universal Ctags 5.9.0(23ea3765), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Jul 16 2021, 21:44:13
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +yaml, +packcc, +optscript
```

### Solution:
...oops, I figured out what ***really*** happened:
* I had previously done `apt-get install ctags`, which was redirected to install exuberant-ctags
* Ran it once, found that it didn't understand Ansible files
* Typed `apt-get remove ctags`, which said: 
```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'exuberant-ctags' instead of 'ctags'
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
```
* In other words, you can install `ctags`, but you can't uninstall `ctags`, you have to uninstall `exuberant-ctags` instead
* Then cloned and installed Universal-ctags in /usr/local/bin/ctags
* Typing 'which ctags' said /usr/local/bin/ctags
* Running 'ctags' *actually* ran /usr/bin/ctags due to bash cache
* Much hilarity ensued as I tried in vain to figure out why ctags was printing the wrong version
* Somehow the cache was cleared by the time I rebuilt universal-ctags and reinstalled it.

### Conclusion:
Anyway, long story short:
* If anyone else stumbles across this problem, type `which -a ctags`, and see if /usr/bin/ctags shows up.
* Apt-get's behaviour is arguably broken
* Bash cache isn't a mind reader; it can't tell if you've installed a new version earlier in your path.  That's what caching is for: so it doesn't have to search your path each time you run something.  Type `hash -r` to clear the cache if you suspect it's the problem.

Live long and prosper!

**Comment by k-takata** [comment]
So, `hash -r` should be enough, right?

**Comment by PenelopeFudd** [comment]
Yes, `hash -r` clears the cache, and if /usr/local/bin/ctags comes first in your $PATH, it wins.

That said, having two versions of ctags on your system is a recipe for disaster; the installation script might want to check for other ctags (e.g. `which -a ctags`) and print a nice big warning.  :-)

**Comment by masatake** [comment]
Could you consider writing your story in https://github.com/universal-ctags/ctags/blob/master/man/ctags-faq.7.rst.in and making a pull request for it?

> That said, having two versions of ctags on your system is a recipe for disaster; the installation script might want to check for other ctags (e.g. which -a ctags) and print a nice big warning. :-)

Pull requests are welcome. I cannot find a good place to add the code for it.

**Comment by masatake** [comment]
b
